cjworkbench/i18n/trans.py

```diff
@@ -90,22 +79,67 @@ def localize_html(
     
     HTML is escaped in the message, as well as in arguments and tag attributes.
     """
-    if locale_id != default_locale:
-        try:
-            return _get_translations(locale_id).get_and_format_html_message(
-                message_id, context=context, arguments=arguments, tags=tags
-            )
-        except ICUError as err:
-            logger.exception(
-                f"Error in po file for locale {locale_id} and message {message_id}: {err}"
-            )
-        except KeyError as err:
-            pass
-    return _get_translations(default_locale).get_and_format_html_message(
-        message_id, context=context, arguments=arguments, tags=tags
+    return MessageLocalizer.for_application_messages(locale_id).localize_html(
+        message_id, arguments=arguments, tags=tags, context=context
     )
 
 
+class MessageLocalizer:
```

Comment:
This class has two functions:

1. It has `.localize()` and `.localize_html()`.
2. Its class acts as a singleton registry of all messages.

Let's break it into two classes. The new class could be `MessageLocalizerRegistry` ... and then we'd have a singleton, `MESSAGE_LOCALIZER_REGISTRY`. Look to `cjwstate/models/module_registry` for inspiration.